ui/src/views/Rules/ListRules/index.jsx

```diff
@@ -454,28 +454,44 @@ function ListRules(props) {
         rule.product || (rule.scheduledChange && rule.scheduledChange.product);
       const ruleScheduledChangeChannel =
         rule.scheduledChange && rule.scheduledChange.channel;
-      let ruleChannel;
+      const ruleChannel = rule.channel;
 
       if (ruleProduct !== productFilter) {
         return false;
       }
 
-      if (ruleScheduledChangeChannel) {
-        if (ruleScheduledChangeChannel.includes('*')) {
-          ruleChannel = rule.channel && ruleScheduledChangeChannel;
-        } else {
-          ruleChannel = ruleScheduledChangeChannel && rule.channel;
+      if (channelFilter) {
+        // if neither the rule nor the scheduled rule's channel is an exact
+        // match for the filter (after stripping away a possible wildcard)
+        // this rule does not match
+        // similarly, if the rule or scheduled rule has a wildcard, and the
+        // selected channel does not match either, this rule does not match
+        let ruleChannelMatches = false;
+        let scChannelMatches = false;
+
+        if (ruleChannel) {
+          if (ruleChannel.indexOf('*') === -1) {
+            if (ruleChannel === channelFilter) {
+              ruleChannelMatches = true;
+            }
+          } else if (channelFilter.startsWith(ruleChannel.split('*')[0])) {
+            ruleChannelMatches = true;
+          }
```

Comment:
Could this be simplified to this?
`const ruleChannelMatches = ruleChannel && ruleChannel.includes(channelFilter);`

Reply:
And then:
`const scChannelMatches = ruleScheduledChangeChannel && ruleScheduledChangeChannel.includes(channelFilter)`